main.js

```diff
@@ -1,78 +1,103 @@
-let DisplayBooks = (index) => {
-  let bgcolor = '';
-  if (savebook.BookData.indexOf(index) % 2 !== 0) {
-    bgcolor = '#fff';
+const getBooks = () => {
+  let books;
+  if (localStorage.getItem('books') === null) {
+    books = [];
   } else {
-    bgcolor = '#ff99';
+    books = JSON.parse(localStorage.getItem('books'));
   }
-  const newBook = document.createElement('div');
-  newBook.classList.add('book-item');
-  newBook.classList.add(bgcolor);
-  newBook.setAttribute('id', index.bookid); newBook.innerHTML = `
- <p>${index.title} <br> written by ${index.author}</p> <hr>`;
-  const removeBook = document.createElement('button');
-  removeBook.innerHTML = 'Remove';
-  removeBook.addEventListener('click', () => savebook.removeBook(index.bookid));
-  newBook.appendChild(removeBook);
-  listOfBooks.appendChild(newBook);
+  return books;
 };
 
-// Add Button
-const addnewBook = document.querySelector('.add-btn');
-addnewBook.addEventListener('click', () => {
-  const item = getformInput();
-  savebook.addBook(item);
-});
+const addBookStr = (book) => {
+  const books = getBooks();
+  books.push(book);
+  localStorage.setItem('books', JSON.stringify(books));
+};
+
+const deleteBookStr = (bookIndex) => {
+  const books = getBooks();
+
+  books.forEach((book, index) => {
+    if (bookIndex === index) {
+      books.splice(index, 1);
+    }
+  });
+  localStorage.setItem('books', JSON.stringify(books));
+};
+
+// Array UI
+const books = getBooks();
+
+const addBook = (book) => {
+  const list = document.querySelector('.book-list');
+  const div = document.createElement('div');
+  div.classList.add('list-container');
+  list.appendChild(div);
+
+  div.innerHTML = `
+  <h3>${book.title}</h3>
+  <p>${book.author}</p>
+  <a href="" class="btn btn-danger btn-sm remove"> Remove </a>
+  <hr> `;
+};
+
+const title = document.querySelector('#title');
+const author = document.querySelector('#author');
+
+const clearFormInputs = () => {
+  title.value = '';
+  author.value = '';
+};
 
-window.onload = () => {
-  savebook.BookData = JSON.parse(localStorage.getItem('AllBooks' || '[]'));
-  if (savebook.BookData === null) {
-    savebook.BookData = [];
-    return;
+const deleteBookList = (element) => {
+  if (element.classList.contains('remove')) {
+    element.parentElement.remove();
   }
-  savebook.BookData.forEach((item) => DisplayBooks(item));
 };
+
+// Events: Display Books
+
+books.forEach((book) => addBook(book));
+
+// Event: add a book
+
+const formBook = document.querySelector('.new-book-container');
+
+formBook.addEventListener('submit', (e) => {
+  e.preventDefault();
+  booksObject.title = title.value;
+  booksObject.author = author.value;
+
+  // validation and Alerts
+  if (title.value === '' || author.value === '') {
+    //
+  } else {
+    // add book to defined array
+    addBook(booksObject);
+
+    // add book to storage
+    addBookStr(booksObject);
+    // clear form
+    clearFormInputs();
+  }
+
+  window.location.reload();
```

Comment:
- [ ] While fixing your application behavior as suggested by the previous reviewer you have added some extra things which are making your application behavior really unprofessional because when you add a book then the page should not reload it would be nice if your application does everything without reloading the page 👍 It would be nice if you remove the `window.location.reload()` because it is making your application reload whenever you create a new book 🤔